protocol/messenger_profile_showcase.go

```diff
@@ -107,6 +109,48 @@ func (m *Messenger) validateCollectiblesOwnership(accounts []*identity.ProfileSh
 	return nil
 }
 
+func (m *Messenger) validateCommunitiesMembership(communities []*identity.ProfileShowcaseCommunity, contactPubKey *ecdsa.PublicKey) ([]*identity.ProfileShowcaseCommunity, error) {
+	validatedCommunities := []*identity.ProfileShowcaseCommunity{}
+
+	for _, communityEntry := range communities {
+		community, err := m.FetchCommunity(&FetchCommunityRequest{
+			CommunityKey:    communityEntry.CommunityID,
+			Shard:           nil,
+			TryDatabase:     true,
+			WaitForResponse: true,
+		})
+		if err != nil {
+			m.logger.Warn("failed to fetch community for profile entry ", zap.Error(err))
```

Comment:
If err is nil, we say that we failed the fetch the community but still we continue some processing after the that error. Isn't this a recipe of random behavior? why shoulndn't we return from here to handle the error?

Reply:
This err can be raised from [this place](https://github.com/status-im/status-go/blob/e99e14dd35c283a992211bd807b527f105418dc6/protocol/messenger_store_node_request_manager.go#L77), which (i believe) can to raise a connection error